src/Microsoft.OData.Client/Serialization/HttpClientRequestMessage.cs

```diff
@@ -71,16 +74,39 @@ public class HttpClientRequestMessage : DataServiceClientRequestMessage, ISendin
         /// <summary>
         /// Constructor for HttpClientRequestMessage.
         /// Initializes the <see cref="HttpClientRequestMessage"/> class.
+        /// </summary>
+        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args)
+            : this(args, null)
+        {
+        }
+
+        /// <summary>
+        /// Constructor for HttpClientRequestMessage that takes a HttpClient instance.
+        /// Initializes the <see cref="HttpClientRequestMessage"/> class.
         /// The args.ActualMethod is the actual method. In post tunneling situations method will be POST instead of the specified verb method.
         /// The args.method is the specified verb method
+        /// The httpClient is an instance of HttpClient that will be used in making requests.
         /// </summary>
-        /// </summary>
-        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args) 
+        public HttpClientRequestMessage(DataServiceClientRequestMessageArgs args, HttpClient client)
             : base(args.ActualMethod)

```

Comment:
I know this isn't actually your change, but if `args` is `null` this will throw a `NullReferenceException`

Reply:
there is already a null check for args before a call to this class is made.

Reply:
I must be missing something, sorry. `HttpClientRequestMessage` is a `public` class, and this is a `public` constructor, so we don't control all callers, right? So it doesn't really matter if some callers check `args` for `null`; if any other callers pass `null`, we will throw `NullReferenceException`. As a minimal repro, consider this code:
```
public class ThrowBase
{
  public ThrowBase(string foo)
  {
  }
}

public class ThrowDerivedArgs
{
  public string Foo { get; set; }
}

public class ThrowDerived : ThrowBase
{
  public ThrowDerived(ThrowDerivedArgs args)
    : base(args.Foo)
  {
    if (args == null)
    {
      throw new ArgumentNullException(nameof(args));
    }
  }
}
```

`new ThrowDerived(null);` will throw `NullReferenceException` instead of `ArgumentNullException`